### PR TITLE
lab_1

### PR DESCRIPTION
**Pull request by lesenia05**

### CODE REVIEW

File: Mobile/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+android {
+    compileSdkVersion 29
+    buildToolsVersion "29.0.2"
+    defaultConfig {
+        applicationId "com.lesenia.mobile"
+        minSdkVersion 21
+        targetSdkVersion 29
+        versionCode 1
+        versionName "1.0"
+        testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
```
Comment (romanvit13) [inline Mobile/app/build.gradle]: I suppose that you didn't write any tests, yes? Consider removing this line. It is unnecessary.

File: Mobile/app/build.gradle [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+            proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    testImplementation 'junit:junit:4.12'
```
Comment (romanvit13) [inline Mobile/app/build.gradle]: This is unnecessary too.

File: Mobile/app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
+        android:roundIcon="@mipmap/ic_launcher_round">
+        <activity android:name="com.lesenia.mobile.UserActivity"
+            android:noHistory="true"
+            android:label="Home"/>
+        <activity android:name="com.lesenia.mobile.LoginActivity"
+            android:noHistory="true"
+            android:label="Log In">
+<!--            <intent-filter>
```
Comment (romanvit13) [inline Mobile/app/src/main/AndroidManifest.xml]: Commented lines in the commit. Remove them or finish your code.

File: Mobile/app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                <action android:name="android.intent.action.MAIN" />
+                <category android:name="android.intent.category.LAUNCHER" />
+            </intent-filter>-->
+        </activity>
+        <activity android:name="com.lesenia.mobile.RegisterActivity"
+            android:noHistory="true"
+            android:label="Sign Up">
```
Comment (romanvit13) [inline Mobile/app/src/main/AndroidManifest.xml]: I suppose the label should match the name of the Activity.

File: Mobile/app/src/main/java/com/lesenia/mobile/LoginActivity.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+        if (isDataValid(email, password)) {
+            Task<AuthResult> authResultTask = auth.signInWithEmailAndPassword(email, password)
+                    .addOnCompleteListener(LoginActivity.this, new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+                            if (task.isSuccessful()) {
+                                startActivity(new Intent(LoginActivity.this, UserActivity.class));
+                            } else {
+                                loginError();
```
Comment (romanvit13) [inline Mobile/app/src/main/java/com/lesenia/mobile/LoginActivity.java]: The name of this method should deliver action, not statement.

File: Mobile/app/src/main/java/com/lesenia/mobile/LoginActivity.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+                            }
+                        }
+                    });
+        }
+    }
+
+    private boolean isEmailValid(final String email) {
+        if (email.isEmpty()) {
+            emailFieldLayout.setError(getString(R.string.enter_email));
```
Comment (romanvit13) [inline Mobile/app/src/main/java/com/lesenia/mobile/LoginActivity.java]: I don't think you have to set the error here. One method - one action.

File: Mobile/app/src/main/java/com/lesenia/mobile/LoginActivity.java [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+            passwordFieldLayout.setError(null);
+            return true;
+        }
+    }
+
+    private boolean isDataValid(final String email, final String password) {
+        boolean emailValid = isEmailValid(email);
+        boolean passwordValid = isPasswordValid(password);
+
+        return emailValid && passwordValid;
```
Comment (romanvit13) [inline Mobile/app/src/main/java/com/lesenia/mobile/LoginActivity.java]: You can simplify: return isEmailValid(email) && isPasswordValid(password).

Comment (lesenia05) [reply Mobile/app/src/main/java/com/lesenia/mobile/LoginActivity.java]: Fixed , thank you!

**Pull request merged**